### PR TITLE
Deploy cWETHv3 to Goerli

### PR DESCRIPTION
**Pull request by kevincheng96**
This PR adds a new `goerli-weth` base and deploy script for the Goerli cWETHv3 deploy.

The deploy script is very similar to the mainnet cWETHv3 script. The main differences are:
- Deploying price feeds for stETH / ETH and cbETH / ETH
- Pulling in extra testnet/cross-chain contracts such as `fauceteer` and `fxRoot` using `fromDep`
- Cloning over `cbETH` from mainnet
- Minting `cbETH` to `fauceteer`

### CODE REVIEW

File: deployments/goerli/weth/configuration.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  "rates": {
+    "supplyKink": 0.8,
+    "supplySlopeLow": 0.0325,
+    "supplySlopeHigh": 0.4,
+    "supplyBase": 0,
+    "borrowKink": 0.8,
+    "borrowSlopeLow": 0.035,
+    "borrowSlopeHigh": 0.25,
+    "borrowBase": 0.015
+  },
```
Comment (kevincheng96) [inline deployments/goerli/weth/configuration.json:20]: TODO: Need to update these rates to the WETH model (which is still under discussion I think)

File: deployments/goerli/weth/deploy.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+  // If we haven't spidered new contracts (which we could before minting, but its slow),
+  //  then the proxy contract won't have the impl functions yet, so just do it explicitly
+  const cbETHProxy = contracts.get('cbETH')!, cbETHImpl = contracts.get('cbETH:implementation')!;
+  const cbETH = cbETHImpl.attach(cbETHProxy.address);
+  await deploymentManager.idempotent(
+    async () => (await cbETH.balanceOf(fauceteer.address)).eq(0),
+    async () => {
+      debug(`Minting 10K cbETH to fauceteer`);
+      const amount = exp(10_000, await cbETH.decimals());
```
Comment (kevincheng96) [inline deployments/goerli/weth/deploy.ts:154]: Is this amount good?

**Comment by scott-silver** [comment]
do we need to add `goerli-weth` to `deploy-market`, `enact-migration` and `prepare-migration`?

**Comment by kevincheng96** [comment]
> do we need to add `goerli-weth` to `deploy-market`, `enact-migration` and `prepare-migration`?

I don't think so since those actually take the deployment as an input (e.g. "usdc").

File: deployments/goerli/weth/deploy.ts (hunk 2 above)
Comment (jflatow) [reply deployments/goerli/weth/deploy.ts:154]: If it doesn't require goerli ETH then maybe a larger number?

Comment (jflatow) [reply deployments/goerli/weth/deploy.ts:154]: If it doesn't require goerli ETH then maybe a larger amount? It should be fine

Comment (kevincheng96) [reply deployments/goerli/weth/deploy.ts]: Ok increased the amount minted to 1mn cbETH. Each fauceteer `drip` would give roughly 100 ETH.

File: deployments/goerli/weth/deploy.ts [hunk 3]
```diff
@@ -150,8 +150,8 @@ async function mintTokens(deploymentManager: DeploymentManager) {
   await deploymentManager.idempotent(
     async () => (await cbETH.balanceOf(fauceteer.address)).eq(0),
     async () => {
-      debug(`Minting 10K cbETH to fauceteer`);
-      const amount = exp(10_000, await cbETH.decimals());
+      debug(`Minting 1M cbETH to fauceteer`);
+      const amount = exp(1_000_000, await cbETH.decimals());
```
Comment (jflatow) [inline deployments/goerli/weth/deploy.ts:154]: 🔥

File: deployments/goerli/weth/configuration.json (hunk 1 above)
Comment (kevincheng96) [reply deployments/goerli/weth/configuration.json]: Done. Updated to the params provided by Gauntlet [here](https://www.comp.xyz/t/initialize-compound-iii-weth-on-ethereum/3737/21). I had to annualize the per-second params they provided.

**Review by jflatow — APPROVED** [review]
💪

**Review by scott-silver — APPROVED** [review]
🎉

File: deployments/goerli/weth/deploy.ts [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+async function deployContracts(deploymentManager: DeploymentManager, deploySpec: DeploySpec): Promise<Deployed> {
+  const ethers = deploymentManager.hre.ethers;
+  const signer = await deploymentManager.getSigner();
+
+  // Declare existing assets as aliases
+  const WETH = await deploymentManager.existing('WETH', '0x42a71137C09AE83D8d05974960fd607d40033499', 'goerli');
+  const wstETH = await deploymentManager.existing('wstETH', '0x4942BBAf745f235e525BAff49D31450810EDed5b', 'goerli');
+
+  // Import shared contracts from cUSDCv3
+  // Note: We need to first spider the contracts on the cUSDC base. Consider moving this spider logic into `fromDep`.
```
Comment (jflatow) [inline deployments/goerli/weth/deploy.ts:24]: In the future we could also have a `fromDep` to return all aliases as object and could destructure ```
const { 
 cometAdmin, 
 cometFactory, 
 'configurator:implementation': $configuratorImpl,
  ...
  } = await deploymentManager.fromDep(network, base)``` and spider from there

File: deployments/goerli/weth/configuration.json (hunk 1 above)
Comment (jflatow) [reply deployments/goerli/weth/configuration.json]: Of course it's JSON so we can't comment on the conversions or write them as expressions 😭

File: deployments/goerli/weth/deploy.ts (hunk 4 above)
Comment (jflatow) [reply deployments/goerli/weth/deploy.ts:24]: Eh, this breaks putAlias, which is why it is the way it is currently

**Comment by kevincheng96** [comment]
Rebased on top of `jflatow/mainnet-eth-live` to get the latest `fromDep` changes so we no longer need to explicitly spider within the deploy script.

File: deployments/goerli/weth/deploy.ts [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+  await mintTokens(deploymentManager);
+  return deployed;
+}
+
+async function deployContracts(deploymentManager: DeploymentManager, deploySpec: DeploySpec): Promise<Deployed> {
+  const ethers = deploymentManager.hre.ethers;
+  const signer = await deploymentManager.getSigner();
+
+  // Declare existing assets as aliases
+  const WETH = await deploymentManager.existing('WETH', '0x42a71137C09AE83D8d05974960fd607d40033499', 'goerli');
```
Comment (scott-silver) [inline deployments/goerli/weth/deploy.ts:20]: looks like there are a number of different instances of the `WETH9` contract on Goerli.

Looks like Uniswap uses `0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6`: https://docs.uniswap.org/contracts/v3/reference/deployments

Comment (kevincheng96) [reply deployments/goerli/weth/deploy.ts:20]: Yeah, for some reason our Goerli USDC deployment has its version of WETH. I chose to use that same WETH to be consistent.

File: deployments/goerli/weth/deploy.ts [hunk 6]
```diff
@@ -115,9 +115,9 @@ async function deployContracts(deploymentManager: DeploymentManager, deploySpec:
     'bulker',
     'bulkers/MainnetBulker.sol',
     [
-      comet.governor(),        // admin_
-      WETH.address,            // weth_
-      wstETH.address           // wsteth_
+      await comet.governor(),        // admin_
```
Comment (jflatow) [inline deployments/goerli/weth/deploy.ts:118]: 👍

**Pull request merged**